src/newSeed/stage3.ts

```diff
@@ -40,6 +40,8 @@ export class Stage3 extends BaseStage {
     if (message) {
       this.validationError(message);
     } else {
+      this.persistData();
```

Comment:
Some stages have a function persistData() that saves some information. This persisData() is previously only called in the detached function of each stage but since this function in the detached function is called only if the next button was clicked, I made sure to call this function upon validation using the next button before proceeding